src/main/java/org/broadinstitute/hellbender/tools/validation/CompareReads.java

```diff
@@ -0,0 +1,129 @@
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.engine.ReadsDataSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+
+import java.util.*;
+
+@DocumentedFeature
+@CommandLineProgramProperties(
+        summary = "Compares the base qualities of two SAM/BAM/CRAM files. The reads in the two files must have " +
+                "exactly the same names and appear in the same order.",
+        oneLineSummary = "Compares the base qualities of two SAM/BAM/CRAM files",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class CompareReads extends GATKTool {
+    @Argument(doc = "If output is given, the tool will return a bam with all the mismatching duplicate groups in the first specified file",
+            shortName = "I1", fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = true)
+    protected String input1;
+
+    @Argument(doc = "If output is given, the tool will return a bam with all the mismatching duplicate groups in the second specified input file",
+            shortName = "I2", fullName = "output2", optional = true)
+    protected String input2;
+
+
+    @Override
+    public void traverse() {
+        List<String> errorMessages = new ArrayList<>();
+
+        try(        ReadsDataSource reads1 = new ReadsDataSource(IOUtils.getPath(input1));
+                    ReadsDataSource reads2 = new ReadsDataSource(IOUtils.getPath(input2));) {
+            final Iterator<GATKRead> it1 = reads1.iterator();
+            final Iterator<GATKRead> it2 = reads2.iterator();
+            while (it1.hasNext() && it2.hasNext()) {
+                final SAMRecord read1 = it1.next().convertToSAMRecord(reads1.getHeader());
+                final SAMRecord read2 = it2.next().convertToSAMRecord(reads2.getHeader());
+                final String eqMessage = readsEqualAllowAddingAttributes(read1, read2);
+                if (eqMessage != null){
+                    errorMessages.add(eqMessage);
+                }
+            }
+            if (it1.hasNext() || it2.hasNext()) {
+                //at least one has no more records (because the while loop is done) and at least one does have records. So we're not equal.
+                throw new GATKException("Not the same number of reads");
+            }
+        }
+        if (!errorMessages.isEmpty()) {
+            System.out.println("Here are the errors generated for this file:");
+            for (String e: errorMessages) {
+                System.out.println("\n"+e);
+            }
+        }
+    }
+
+    /**
+     * Compares the reads but ignores order of attributes.
+     * Also allows actualRead to have a superset of attributes of expectedRead.
+     */
```

Comment:
Is the superset of attributes something you want?  It doesn't mention in the doc that they're not treated the same way.